consensus/propagation/reactor.go

```diff
@@ -101,6 +107,15 @@ func NewReactor(self p2p.ID, tracer trace.Tracer, store *store.BlockStore, mempo
 
 type ReactorOption func(*Reactor)
 
+// SetProposalValidator sets the proposal stateful validation function.
+func (blockProp *Reactor) SetProposalValidator(validator validateProposalFunc) {
+	blockProp.proposalValidator = validator
+}
```

Comment:
the way we currently communicate between the consensus state and the propagation reactor make me :see_no_evil: lol

Reply:
this is trying to make the changes the less intrusive. The problem is the mutexes... If we have more access to the state, and we block for reading/writing, we might deadlock since the blocks are all over the repo. So, I guess this is the easiest way to avoid all that